### PR TITLE
Tukaiz develop

### PR DESCRIPTION
**Pull request by francirp**

### CODE REVIEW

File: app/views/locomotive/previews/_toolbar.html.erb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+body.previews-controller  a#reject-link {
+  color: red !important;
+  margin-left: 50px;
+}
+
+body.previews-controller form {
+  float: left;
+}
```
Comment (francirp) [inline app/views/locomotive/previews/_toolbar.html.erb:40]: I hate having to put the styles into the HTML file but can't figure out how to get the assets to precompile correctly to pull them in. Just going to have to ride with this for now. Below is the SCSS that I hope to eventually switch to:

``` css.scss
body.previews-controller {
  margin-top: 0px;
  .fixed {
    top: 60px !important;
  }
  .preview-toolbar {
    height: 60px;
    width: 100%;
    background-color: white;
    position: fixed;
    top: 0;
    z-index: 100;
    .button {
      margin-top: 8px;
      margin-left: 50px;
      height: 40px;
      padding-top: 12px;
    }
    .back-button {
      margin-left: 50px;
    }
  }
  .toolbar-margin {
    height: 60px;
    margin-bottom: 45px;
  }

  a#reject-link {
    color: red !important;
    margin-left: 50px;
  }

  form {
    float: left;
  }

}
```

**Comment by francirp** [comment]
Merging to push to production before EOD

File: app/cells/locomotive/settings_menu_cell.rb [hunk 2]
```diff
@@ -8,7 +8,8 @@ def build_list
       add :theme_assets,  url: theme_assets_path
       add :translations,  url: translations_path
       add :account,       url: edit_my_account_path
+      add :approvals,     url: previews_pages_path
```
Comment (bhennes2) [inline app/cells/locomotive/settings_menu_cell.rb:11]: How are we ensuring only admins can see a link to approvals?

File: app/controllers/locomotive/previews_controller.rb [hunk 3]
```diff
@@ -1,22 +1,65 @@
...
+        @preview.page_id = @page.id
+        @preview.account_id = current_locomotive_account.id
+        @preview.save
+      end
+      redirect_to edit_page_path(@page._id)
+    end
 
+    def update
+      @preview = current_site.previews.find(params[:id])
+      return destroy if params[:commit] == Locomotive::Preview::REJECT_TEXT
```
Comment (bhennes2) [inline app/controllers/locomotive/previews_controller.rb:42]: Do we want to delete a preview if it has been rejected or just mark a flag that says "Rejected"?

File: app/models/locomotive/page.rb [hunk 4]
```diff
@@ -63,6 +63,12 @@ class Page
 
     ## methods ##
 
+    def render(context, options = {})
+      output = super(context)
+      output.gsub!("<body>", options[:toolbar]) if options[:toolbar]
+      return output
```
Comment (bhennes2) [inline app/models/locomotive/page.rb:69]: You don't necessarily need this line.

File: app/views/locomotive/previews/_toolbar.html.erb (hunk 1 above)
Comment (bhennes2) [inline app/views/locomotive/previews/_toolbar.html.erb:40]: Good point.

Can the CSS just go with the other Engine CSS? Maybe as an override?

**Pull request merged**